dependency-convergence/src/main/java/ArtifactData.java

```diff
@@ -0,0 +1,263 @@
+
+      if (metadata.getVersioning() == null) {
+        return new LatestMetadata("", "");
+      }
+
+      String latestVersion;
+      if (metadata.getVersioning().getLatest() == null) {
+        latestVersion = "";
+      } else {
+        latestVersion = metadata.getVersioning().getLatest();
+      }
+
+      String lastUpdated = metadata.getVersioning().getLastUpdated();
+      if (lastUpdated != null && !lastUpdated.isEmpty()) {
+        Date date = dateFormat.parse(lastUpdated);
+        lastUpdated = outputFormat.format(date);
+      } else {
+        lastUpdated = "";
+      }
+      return new LatestMetadata(latestVersion, lastUpdated);
+    } catch (XmlPullParserException | ParseException | IOException ignored) {
+      return new LatestMetadata("", "");
+    }
+  }
+
+  /**
+   * Grabs the Github URL from the POM's scm section. If the URL is not found, then this checks the
+   * associated parent POM (given that the current POM is not already the parent of our artifact)
+   *
+   * @param pomUrl the Maven URL of the artifact's POM file
+   * @return Artifact's Github URL
+   */
+  private static String getScmGithubUrl(String pomUrl) {
+    try {
+      File pomFile = File.createTempFile("pomFile", ".xml");
+      pomFile.deleteOnExit();
+      URL url = new URL(pomUrl);
+      BufferedInputStream input = new BufferedInputStream(url.openStream());
+      FileUtils.copyInputStreamToFile(input, pomFile);
+      MavenXpp3Reader read = new MavenXpp3Reader();
+      Model model = read.read(new FileInputStream(pomFile));
+      if (model == null) {
+        return null;
+      }
+      return model.getUrl();
+    } catch (XmlPullParserException | IOException ignored) {
+    }
+    return null;
+  }
+
+  private static String generatePomFileUrl(Artifact artifact) {
+    String groupPath = artifact.getGroupId().replace('.', '/');
+    return basePath + "/" + groupPath
+        + "/" + artifact.getArtifactId()
+        + "/" + artifact.getVersion()
+        + "/" + artifact.getArtifactId() + "-" + artifact.getVersion() + ".pom";
+  }
+
+  private static String generateMetadataUrl(Artifact artifact) {
+    String groupPath = artifact.getGroupId().replace('.', '/');
+    return basePath + "/" + groupPath
+        + "/" + artifact.getArtifactId()
+        + "/maven-metadata.xml";
+  }
+
+  private static class LatestMetadata {
+
+    String latestVersion, lastUpdated;
+
+    LatestMetadata(String latestVersion, String lastUpdated) {
+      this.latestVersion = latestVersion;
+      this.lastUpdated = lastUpdated;
+    }
+  }
+
+  private static String getSharedDependenciesVersionFromUrl(String pomUrl) {
+    if (pomUrl == null) {
+      return null;
+    }
+    try {
+      File pomFile = File.createTempFile("pomFile", ".xml");
+      pomFile.deleteOnExit();
+      URL url = new URL(pomUrl);
+      BufferedInputStream input = new BufferedInputStream(url.openStream());
+      FileUtils.copyInputStreamToFile(input, pomFile);
```

Comment:
Files.copy form Guava

Reply:
Unfortunately I was finding it a bit difficult to get proper output with this (there's no method for copying an InputStream direct-to-file unfortunately). However, I was able to remove the use of the BufferedInputStream and copy direct from URL